features/step_definitions/hooks.js

```diff
@@ -0,0 +1,22 @@
+const {newSession} = require('w3c-webdriver');
+const {defineSupportCode} = require('cucumber');
+const chromedriver = require('chromedriver');
+const testApp = require('../../test-app');
+const sessionStorage = require('./session-storage');
+
+defineSupportCode(({registerHandler}) => {
+    registerHandler('BeforeFeatures', async () => {
+        chromedriver.start();
+        await testApp.start();
+        sessionStorage.create(await newSession('http://localhost:9515', {
+            browserName: 'Chrome'
+        }));
+    });
+
+    registerHandler('AfterFeatures', async () => {
+        const session = await sessionStorage.get();
```

Comment:
hmm interesting, as I debugged getting the session is an async process, after pinging the chrome driver we return the API binded session object. but here we have to wait for the webdriver to answer.